### DISCUSSION TITLE
React Component Library - How?

### DISCUSSION THREAD

**Discussion by ShaneYu** [web-infra-dev/rsbuild#3273] [Q&A] [answered]
I love how simple it has been to use `RSBuild` to build a React application and it just works, however I am struggling to work out how to use `RSBuild` to build a React Components Library.

In Vite I would set the lib options to achieve this, but I cannot work out how to do this for `RSBuild`, any help would be greatly appreciated.

```javascript
    lib: {
      // Could also be a dictionary or array of multiple entry points.
      entry: "src/lib/index.ts",
      name: "design-system",
      fileName: "index",
      // Change this to the formats you want to support.
      // Don't forget to update your package.json as well.
      formats: ["es", "cjs"],
    },
```

Any idea how to do this with RSBuild? Are there any examples anywhere?

**Comment by chenjiahan** [comment, marked as answer]
See https://github.com/web-infra-dev/rsbuild/discussions/3020

**Reply by ShaneYu** [reply]
Thank you, I will take a look - it sounds like what I need. Was getting very lost trying to work it out. Cheers
